lib/_http_outgoing.js

```diff
@@ -602,6 +603,24 @@ OutgoingMessage.prototype.getHeaderNames = function getHeaderNames() {
 };
 
 
+// Returns an array of the names of the current outgoing raw headers.
+OutgoingMessage.prototype.getRawHeaderNames = function getRawHeaderNames() {
+  const headersMap = this[kOutHeaders];
+  const headers = [];
```

Comment:
Given that we know how long this array is going to be, we should be using `Array(n)` instead.